File: app/src/main/kotlin/thanh/ha/ui/customTag/SpannableClickOverlay.java
package thanh.ha.ui.customTag;

import android.text.style.ClickableSpan;
import android.view.View;

import androidx.annotation.NonNull;


public class SpannableClickOverlay extends ClickableSpan {

    private final SpannableClickAction clickAction;

    public SpannableClickOverlay(@NonNull SpannableClickAction clickAction) {
        this.clickAction = clickAction;
    }

    @Override
    public void onClick(View view) {
        clickAction.onClick();
    }
}

File: app/src/main/kotlin/thanh/ha/ui/customSpanable/SpannableClickOverlay.kt
package thanh.ha.ui.customSpanable

import android.text.style.ClickableSpan
import android.view.View


class SpannableClickOverlay(private val clickAction: SpannableClickAction,
                            val text: String) : ClickableSpan() {

    override fun onClick(view: View) {
        clickAction.onClick(text)
    }
}

